src/Type/WPObjectType.php

```diff
@@ -60,6 +60,8 @@ public function __construct( $config ) {
 		 */
 		do_action( 'graphql_wp_object_type', $config, $this );
 
+		PostStatusRegister::register_status_enum_type( $config['name'] );
```

Comment:
I don't like this being in the constructor - it feels wrong. I am not sure where else in the call stack it can go. It needs to be registered when the type is registered - so perhaps in `TypeRegistry::prepare_type()`? Maybe in the switch statement that invokes this constructor?